### PR TITLE
feat(styles): table fixed end columns, fixed footer and refactoring

### PR DESCRIPTION
**Pull request by platon-rov**
## Related Issue

Closes https://github.com/SAP/fundamental-styles/issues/3302
Refers to https://github.com/SAP/fundamental-ngx/issues/9030
Refers to https://github.com/SAP/fundamental-styles/issues/3962

*Note: it's not a complete v1 preparations, here was made some refactoring to decrease component's complexity*

## Description

* feat: fixed columns on the end (ride side) of the table
* feat: fixed footer for the table
* chore: refactoring styles of the table
* fix: Minor issues fixes

### Before:

Pop-in active state
image

Pop-in selected active state
image

### After:

New features
image

Pop-in active state
image

Pop-in selected active state
image

### CODE REVIEW

File: packages/styles/src/mixins/table/_table-group-and-tree.scss [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+        &--expand {
+          cursor: pointer;
+          white-space: nowrap;
+
+          @include fd-table-nested-cell-padding(20, true);
+        }
+
+        .#{$block}__expand {
+          width: var(--fd-table-header-cell-height);
```
Comment (N1XUS) [inline packages/styles/src/mixins/table/_table-group-and-tree.scss:67]: `--fd-table-header-cell-height` and `--fd-table-cell-height` text format is different than we usualy use:
`--fdTable_Header_Cell_Height`

**Comment by platon-rov** [comment]
> Shouldn't these be handles in RTL? Screenshot 2023-01-31 at 4 15 37 PM

I don't think so, it's from markup where user should handle it, we don't "create" these icons with CSS.

Additionally, we would break the self-containing rule by changing icon styles in the table component.

File: packages/styles/src/mixins/table/_table-fixed.scss [hunk 2]
```diff
@@ -0,0 +1,74 @@
+@import "../mixins";
+
+@mixin fd-table-fixed() {
```
[the remark this answers is not in the document]
Comment (platon-rov) [reply packages/styles/src/mixins/table/_table-fixed.scss:3]: You mean `@extend %placeholder-name`? Unfortunately, then we won't be able to use `$block` in these files.

File: packages/styles/src/mixins/table/_table-group-and-tree.scss [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        @include fd-rtl() {
+          padding-left: 0.5rem;
+          padding-right: $second-level-padding + $third-level-padding + (($i - 2) * $level-padding) + $expand-padding;
+        }
+      }
+    }
+  }
+}
+
+@mixin fd-table-group-and-tree() {
```
[the remark this answers is not in the document]
Comment (platon-rov) [reply packages/styles/src/mixins/table/_table-group-and-tree.scss:44]: same? https://github.com/SAP/fundamental-styles/pull/4214#discussion_r1093607864

File: packages/styles/src/mixins/table/_table-no-borders.scss [hunk 4]
```diff
@@ -0,0 +1,98 @@
+@import "../mixins";
+
+@mixin fd-table-no-borders() {
```
[the remark this answers is not in the document]
Comment (platon-rov) [reply packages/styles/src/mixins/table/_table-no-borders.scss:3]: same? https://github.com/SAP/fundamental-styles/pull/4214#discussion_r1093607864

**Comment by InnaAtanasova** [comment]
> https://deploy-preview-4214--fundamental-styles.netlify.app



> 



> > Shouldn't these be handles in RTL?
> > Screenshot 2023-01-31 at 4 15 37 PM
> 
> I don't think so, it's from markup where user should handle it, we don't "create" these icons with CSS.
> 
> Additionally, we would break the self-containing rule by changing icon styles in the table component.

Actually, not quite true. We need to handle the RTL where possible (doesn't require JS). You can't break self contained because the navigation indicators are handled in Table (they have classes)
Screenshot 2023-02-03 at 12 37 58 PM

**Comment by g-cheishvili** [comment]
In new chromatic baselines for some reason navigation indication icons are rotated.
https://deploy-preview-4214--fundamental-styles.netlify.app/?path=/docs/components-table--primary#navigation-from-table-rows

**Comment by InnaAtanasova** [comment]
I see changes in the markup. Have you documented the Breaking changes?

**Comment by InnaAtanasova** [comment]
The navigation indicators are still not fixed 
Screenshot 2023-02-14 at 3 52 00 PM
Screenshot 2023-02-14 at 3 51 51 PM

**Comment by platon-rov** [comment]
> The navigation indicators are still not fixed

What a pity mistake, fixed, thanks :)

**Comment by InnaAtanasova** [comment]
Should these look like this?
Screenshot 2023-02-15 at 3 04 32 PM

**Comment by platon-rov** [comment]
> Should these look like this? Screenshot 2023-02-15 at 3 04 32 PM

Actually it's a button's issue, I created a ticket https://github.com/SAP/fundamental-styles/issues/4258.

I don't think it's reasonable to fix this in table as we may have navigation buttons like these in many places, it would be better to add a special type of button which will mirror its icon in rtl.

**Review by InnaAtanasova — APPROVED** [review]

**Pull request merged**